### DISCUSSION TITLE
package rotation>dims box selection

### DISCUSSION THREAD

**Discussion by bobatlanta** [dvdoug/BoxPacker#320] [Q&A]
hi Doug we're considering moving from volume-based dims shipping app to BoxPacker - i believe volume-based app we use now uses smallest volume dims box that (a) has volume equal to/greater than parcel being packed, and (b) has length equal to/greater than length of parcel being packed

i understand BoxPacker algorithm packs differently - re 20 or so dims boxes we created to test parcel rotation and dims box being used for packing, BoxPacker skips some dims boxes that volume based app would otherwise use i.e.. (a) & (b) methodology above

besides documenation>principles of operation, is there a link that explains BoxPacker package rotation & dims box selection (sorry, i'm not a developer or strong technical person), but i want to try t figure it out - some posts i looked at this forum indicate your algorithm more accurate than volume-based

thanks very much!

**Comment by dvdoug** [comment]
Hi @bobatlanta 

There's nothing formal written down other than that page you found - the algorithm is a set of heuristics that I've developed. It won't necessary give you the "smartest" packing each time (no algorithm exists that can do that), but it does operate in actual, physical space - if it says that x items fit in y boxes then the items do fit.

**Comment by bobatlanta** [comment]
Hi Doug, thanks for getting back to me & info below, I’ll re-read pg 6 Boxpacker document again

 

In meantime, going to ask if you can answer below, it will help me understand the algorithm at high level, & if not easy to answer/solve below, understand & no problem

 

Re parcel test shipment w/dims (L, W, H) of 20, 10, 10, & comparing to dims boxes below I created in Calcurates, algorithm packs test parcel in the 3080v box, even though 2376v box (a) has more volume than test parcel, & (b) has length longer than parcel length – I think these are the main two “if-then” arguments that shipperhq (i.e. volume based & app we’re using now) explained to me several years ago when I was creating my dims boxes

 



 

Re pg 6 Boxpacker documentation, i believe volume comparison is still the first “if-then” argument, same as volume-based, however, what are the next “if-then” arguments that BP heuristic algorithm uses that cause it to bypass 2376v dims box and instead pack test parcel in 3080v box

 

again, if no easy answer to my question, understand & thank you again for replying to my post

 

thanks Doug!

 

Bob Orkand

Atlanta GA

404 422 3287

**Comment by dvdoug** [comment]
Hi @bobatlanta 

I'm not sure if maybe there was some kind of graphic you meant to include (I see you replied via email not directly on GitHub), I can't see any details of the example packing you mention so I'm unable to answer any specifics about how BoxPacker deals with it.

But assuming this was an example where all of the items would fit into a single box, then yes the one with the smallest volume should be picked

**Comment by bobatlanta** [comment]
Hi Doug, I reread Boxpacker documentation & figured out most of what I was looking for i.e. high level if/then arguments for package rotation into dims box – it worked for most of dimensional package testing I did, & will help me fine-tune dimensional box sizes 

Thanks!
